cloud/scope/machine.go

```diff
@@ -338,6 +338,9 @@ func (m *MachineScope) InstanceSpec(log logr.Logger) *compute.Instance {
 				m.ClusterGetter.Name(),
 			),
 		},
+		Params: &compute.InstanceParams{
```

Comment:
This will tag the compute instances.

We should probably tag other GCP resources we are creating, like the actual GKE cluster:
https://cloud.google.com/kubernetes-engine/docs/how-to/tags

So we probably need to call [CreateTagBinding](https://cloud.google.com/go/docs/reference/cloud.google.com/go/resourcemanager/latest/apiv3#cloud_google_com_go_resourcemanager_apiv3_TagBindingsClient_CreateTagBinding) for other resources. wdyt?

Reply:
Disks attached to Instance also supports Tags, I think we should include tags in disk params at [[1]](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/c0eb2c2901611d75e1adabf854fbb75353bf3ebf/cloud/scope/machine.go#L241) and [[2]](https://github.com/kubernetes-sigs/cluster-api-provider-gcp/blob/c0eb2c2901611d75e1adabf854fbb75353bf3ebf/cloud/scope/machine.go#L255)?